### PR TITLE
feat: support for `ALTER/DROP SUSCRIPTION` 

### LINKED ISSUE #246: Add support for `ALTER SUBSCRIPTION mysub ENABLE/DISABLE`
https://www.postgresql.org/docs/current/sql-altersubscription.html

### PR DESCRIPTION
**Pull request by TianyuZhang1214**
**Resolve #246**

### New Features
- **ALTER SUBSCRIPTION**:  
  Added support for `ALTER SUBSCRIPTION mysub [ENABLE | DISABLE]`.
  - `DISABLE`: Pauses replication from the source PostgreSQL.
  - `ENABLE`: Resumes replication from the source PostgreSQL.

- **DROP SUBSCRIPTION**:  
  Added support for `DROP SUBSCRIPTION mysub`.  
  *Note*: Dropping a subscription does not remove any previously replicated data.

### Improvements
- Introduced a `subscriptionMap` in `subscription.go` to manage all subscriptions centrally.
- Merged system table `__sys__.pg_replication_lsn` into `__sys__.pg_subscription`.

### CODE REVIEW

File: catalog/internal_tables.go [hunk 1]
```diff
@@ -120,19 +176,12 @@ var InternalTables = struct {
...
-		DDL:          "slot_name TEXT PRIMARY KEY, lsn TEXT",
-	},
 	PgSubscription: InternalTable{
 		Schema:       "__sys__",
 		Name:         "pg_subscription",
-		KeyColumns:   []string{"name"},
-		ValueColumns: []string{"connection", "publication"},
-		DDL:          "name TEXT PRIMARY KEY, connection TEXT, publication TEXT",
+		KeyColumns:   []string{"subname"},
+		ValueColumns: []string{"subconninfo", "subpublication", "subskiplsn", "subenabled"},
```
Comment (fanyang01) [inline catalog/internal_tables.go:183]: The `sub` prefix for all column names can be removed.

File: pgserver/connection_handler.go [hunk 2]
```diff
@@ -1059,15 +1059,15 @@ func (h *ConnectionHandler) handledPSQLCommands(statement string) (bool, error)
...
-		query, err := h.convertQuery(`select d.datname as "Name", 'postgres' as "Owner", 'UTF8' as "Encoding", 'en_US.UTF-8' as "Collate", 'en_US.UTF-8' as "Ctype", 'en-US' as "ICU Locale", case d.datlocprovider when 'c' then 'libc' when 'i' then 'icu' end as "locale provider", '' as "access privileges" from pg_catalog.pg_database d order by 1;`)
+		query, err := h.convertQuery(`select d.datname as "Subscription", 'postgres' as "Owner", 'UTF8' as "Encoding", 'en_US.UTF-8' as "Collate", 'en_US.UTF-8' as "Ctype", 'en-US' as "ICU Locale", case d.datlocprovider when 'c' then 'libc' when 'i' then 'icu' end as "locale provider", '' as "access privileges" from pg_catalog.pg_database d order by 1;`)
```
Comment (fanyang01) [inline pgserver/connection_handler.go:1062]: Accidental changes?

File: pgserver/logrepl/subscription.go [hunk 3]
```diff
@@ -1,35 +1,136 @@
...
+		}
+		tempMap[name] = &Subscription{
+			Subscription: name,
+			Conn:         conn,
+			Publication:  pub,
+			LsnStr:       lsn,
+			Enabled:      enabled,
+			Replicator:   nil,
+		}
+	}
```
Comment (fanyang01) [inline pgserver/logrepl/subscription.go:54]: Check `rows.Err()` after the for loop.

File: pgserver/logrepl/subscription.go [hunk 4]
```diff
@@ -1,35 +1,136 @@
...
 
-func FindReplication(db *stdsql.DB) (name, conn, pub string, ok bool, err error) {
-	var rows *stdsql.Rows
-	rows, err = db.QueryContext(context.Background(), catalog.InternalTables.PgSubscription.SelectAllStmt())
+var keyColumns = []string{"subname"}
+var statusValueColumns = []string{"subenabled"}
+var lsnValueColumns = []string{"subskiplsn"}
+
+var subscriptionMap = sync.Map{}
+var mu sync.Mutex
```
Comment (fanyang01) [inline pgserver/logrepl/subscription.go:28]: Unnecessary and dangerous global mutex.

File: pgserver/logrepl/subscription.go [hunk 5]
```diff
@@ -1,35 +1,136 @@
...
+	rows, err := adapter.QueryCatalog(ctx, catalog.InternalTables.PgSubscription.SelectAllStmt())
 	if err != nil {
-		return
+		return err
 	}
 	defer rows.Close()
 
-	if !rows.Next() {
-		return
+	var tempMap = make(map[string]*Subscription)
```
Comment (fanyang01) [inline pgserver/logrepl/subscription.go:39]: The `temp` prefix is unnecessary. It is a local variable, implying it is temporary.

File: pgserver/subscription_handler.go [hunk 6]
```diff
@@ -3,102 +3,221 @@ package pgserver
...
+		return h.executeAlterDisable(subscriptionConfig)
+	default:
+		return fmt.Errorf("unsupported action: %s", subscriptionConfig.Action)
 	}
+}
 
-	subscriptionName := subscriptionMatch[1]
-	connectionString := subscriptionMatch[2]
-	publicationName := subscriptionMatch[3]
+func (h *ConnectionHandler) executeAlterEnable(subscriptionConfig *SubscriptionConfig) error {
```
Comment (fanyang01) [inline pgserver/subscription_handler.go:178]: `executeAlterEnable` -> `executeEnableSubscription(config *SubscriptionConfig)`

File: pgserver/subscription_handler.go [hunk 7]
```diff
@@ -3,102 +3,221 @@ package pgserver
...
 	}
 
-	// Initialize SubscriptionConfig struct
-	config := &SubscriptionConfig{
-		SubscriptionName: subscriptionName,
-		PublicationName:  publicationName,
+	return commitAndUpdate(sqlCtx)
+}
+
+func (h *ConnectionHandler) executeAlterDisable(subscriptionConfig *SubscriptionConfig) error {
```
Comment (fanyang01) [inline pgserver/subscription_handler.go:192]: -> `executeDisableSubscription(config *SubscriptionConfig)`

File: catalog/internal_tables.go (hunk 1 above)
Comment (TianyuZhang1214) [reply catalog/internal_tables.go:183]: `sub` is chosen to maintain consistency with the [pg_subscription catalog in PostgreSQL](https://www.postgresql.org/docs/current/catalog-pg-subscription.html).

File: pgserver/connection_handler.go (hunk 2 above)
Comment (TianyuZhang1214) [reply pgserver/connection_handler.go]: fixed

File: pgserver/logrepl/subscription.go (hunk 3 above)
Comment (TianyuZhang1214) [reply pgserver/logrepl/subscription.go:51]: fixed

File: pgserver/logrepl/subscription.go (hunk 4 above)
Comment (TianyuZhang1214) [reply pgserver/logrepl/subscription.go]: fixed

File: pgserver/subscription_handler.go (hunk 6 above)
Comment (TianyuZhang1214) [reply pgserver/subscription_handler.go]: fixed

File: pgserver/subscription_handler.go (hunk 7 above)
Comment (TianyuZhang1214) [reply pgserver/subscription_handler.go]: fixed

**Pull request merged**